### PR TITLE
CORCI-27 test: Fix a number of spelling mistakes.

### PR DESCRIPTION
**Pull request by ashleypittman**
Fix a number of spelling mistakes in the codebase as picked up by codespell.

### CODE REVIEW

File: doc/user/posix.md [hunk 1]
```diff
@@ -204,7 +204,7 @@ library works in conjunction with DFuse and allows the interception of POSIX I/O
 calls and issue the I/O operations directly from the application context through
 libdaos without any application changes.  This provides kernel-bypass for I/O data
 leading to improved performance.
-To use this set the LD_PRELOAD to point to the shared library in the DOAS install
+To use this set the LD_PRELOAD to point to the shared library in the DOES install
```
Comment (ashleypittman) [inline doc/user/posix.md:207]: This one is incorret.

File: src/cart/test/launch.py [hunk 2]
```diff
@@ -63,8 +63,8 @@ def printhelp():
     print("\n")
     print("Usage: ./launch <tag>\n")
     print("\ttag:  ")
-    print("\tall\tRun all tests ")
-    print("\tproto\tRun basic CaRT protocol test ")
+    print("\tall\turn all tests ")
+    print("\tproto\turn basic CaRT protocol test ")
```
Comment (ashleypittman) [inline src/cart/test/launch.py:67]: These are wrong

File: src/client/dfuse/dfuse_vector.c [hunk 3]
```diff
@@ -31,7 +31,7 @@
 #include "dfuse_obj_da.h"
 #include "dfuse_vector.h"
 
-#define CAS(valuep, old, new) \
+#define CASE(valuep, old, new) \
```
Comment (ashleypittman) [inline src/client/dfuse/dfuse_vector.c:34]: This is wrong

File: src/client/java/daos-java/src/main/java/io/daos/dfs/DaosFsClient.java [hunk 4]
```diff
@@ -1076,7 +1076,7 @@ public synchronized int getRefCnt() {
    * <p>
    * Please note that new pool and new container will be created if their ids (poolId and containerId) are {@code null}.
    */
-  public static class DaosFsClientBuilder implements Cloneable {
+  public static class DaosFsClientBuilder implements Clonable {
```
Comment (ashleypittman) [inline src/client/java/daos-java/src/main/java/io/daos/dfs/DaosFsClient.java:1079]: I don't know if this one is correct.

File: src/control/server/config_test.go [hunk 5]
```diff
@@ -358,7 +358,7 @@ func TestServer_ConfigRelativeWorkingPath(t *testing.T) {
 		expErrMsg string
 	}{
 		"path exists":       {inPath: "uncommentedDefault"},
-		"path doesnt exist": {expErrMsg: "no such file or directory"},
+		"path does not exist": {expErrMsg: "no such file or directory"},
```
Comment (ashleypittman) [inline src/control/server/config_test.go:361]: This change is in a variable so I'm not sure of the effect, but it is test code.

File: utils/sl/prereq_tools/c_source/unit_test.c [hunk 6]
```diff
@@ -41,7 +41,7 @@ static int mocked_fn_cnt;
 bool verbose_unit_test_output = FALSE;
 
 /*
- * Mimick requires some functions setup that are otherwise optional, but doesn't
+ * Mimic requires some functions setup that are otherwise optional, but doesn't
```
Comment (ashleypittman) [inline utils/sl/prereq_tools/c_source/unit_test.c:44]: I don't know if this one is correct or not.

**Comment by ashleypittman** [comment]
This needs further work to either resolve or whitelist the Java error, but other than that it's ready to lank I think.

File: doc/user/posix.md [hunk 7]
```diff
@@ -204,7 +204,7 @@ library works in conjunction with DFuse and allows the interception of POSIX I/O
 calls and issue the I/O operations directly from the application context through
 libdaos without any application changes.  This provides kernel-bypass for I/O data
 leading to improved performance.
-To use this set the LD_PRELOAD to point to the shared library in the DOAS install
+To use this set the LD_PRELOAD to point to the shared library in the DAOS install
```
Comment (brianjmurrell) [inline doc/user/posix.md:207]: How does it know how to spell _DAOS_?

File: utils/docker/Dockerfile.centos.7 [hunk 8]
```diff
@@ -79,7 +79,8 @@ RUN yum -y install \
 #  - excluding mpi4py as it depends on CORCI-635
 RUN pip3 install avocado-framework; \
     pip3 install avocado-framework-plugin-result-html; \
-    pip3 install avocado-framework-plugin-varianter-yaml-to-mux
+    pip3 install avocado-framework-plugin-varianter-yaml-to-mux; \
+    pip3 install codespell
```
Comment (brianjmurrell) [inline utils/docker/Dockerfile.centos.7:83]: Can you open a ticket to create a codespell RPM.  It's already packaged for Fedora so an EL7 package should be trivial.

Probably, really, we should be doing our linting, etc. on Fedora anyway.  It's tools are typically newer.

File: doc/user/posix.md (hunk 7 above)
Comment (ashleypittman) [reply doc/user/posix.md:207]: It doesn't, but I do.

File: utils/docker/Dockerfile.centos.7 (hunk 8 above)
Comment (ashleypittman) [reply utils/docker/Dockerfile.centos.7:83]: Is that necessary?  I think the Docker stuff was fairly efficient at caching steps so what would a RPM buy us?

Comment (brianjmurrell) [reply utils/docker/Dockerfile.centos.7:83]: It buys us predictability and reproducibility.  `pip install` does not install any particular version and will always just install the latest.  Installing RPMs installs a known/tested version or at minimum documents the version being installed.

File: doc/user/posix.md (hunk 7 above)
Comment (brianjmurrell) [reply doc/user/posix.md:207]: But if it doesn't know how to spell it, how does it know it's not a spelling error?

Comment (ashleypittman) [reply doc/user/posix.md:207]: Ah I see.  codespell checks for common typos, not against a dictionary.  The real advantage it has is that it can automatically correct things a lot of the time.

**Comment by ashleypittman** [comment]
> It failed checkpatch on it's latest run:
> 
> ```
> [2020-05-06T15:30:41.159Z] + codespell --ignore-words-list nd,uint,ths,ba,creat,te,cas,mmap,pres,crashers,dout,tre,reord,mimick --skip '.git,*.png,*.PNG,*.pyc,./src/rdb/raft/*,./src/control/vendor/*,./.sconf_temp/*,.sconsign.dblite,RSA.golden,./code_review/*'
> [2020-05-06T15:30:43.148Z] ./src/client/java/daos-java/src/main/java/io/daos/dfs/DaosFsClient.java:1079: Cloneable  ==> Clonable
> ```

Yes, someone who understands what that Java code is doing needs to look at that.  I suspect it's part of the published API which means we either accept it whitelist it, or we fix it any update whatever other packages we have that use it.  I'm not sure who that person would be though.

File: utils/docker/Dockerfile.centos.7 (hunk 8 above)
Comment (ashleypittman) [reply utils/docker/Dockerfile.centos.7:83]: True.  I'll file a ticket then.

Comment (brianjmurrell) [reply utils/docker/Dockerfile.centos.7:83]: Can you just let me know the ticket number when you do so that I can make sure it's on our radar?

**Comment by brianjmurrell** [comment]
> Yes, someone who understands what that Java code is doing needs to look at that. I suspect it's part of the published API which means we either accept it whitelist it, or we fix it any update whatever other packages we have that use it. I'm not sure who that person would be though.

Looks like it was added in 51886d9cb2bdd5ca2c3ec6f817cdf480f6a9e7fc (#1714) by @zjf2012.  Perhaps he/she could comment.

That said, this looks like some kind of Java language construct: https://beginnersbook.com/2015/01/cloneable-interface-in-java-object-cloning/ so maybe the only thing to do is whitelist it.

Also, you now have a merge conflict in `src/control/server/ctl_harness.go`.

File: doc/user/posix.md (hunk 7 above)
Comment (brianjmurrell) [reply doc/user/posix.md:207]: Ahhh.  I see.  Interesting.

**Comment by zjf2012** [comment]
> > Yes, someone who understands what that Java code is doing needs to look at that. I suspect it's part of the published API which means we either accept it whitelist it, or we fix it any update whatever other packages we have that use it. I'm not sure who that person would be though.
> 
> Looks like it was added in [51886d9](https://github.com/daos-stack/daos/commit/51886d9cb2bdd5ca2c3ec6f817cdf480f6a9e7fc) (#1714) by @zjf2012. Perhaps he/she could comment.
> 
> That said, this looks like some kind of Java language construct: https://beginnersbook.com/2015/01/cloneable-interface-in-java-object-cloning/ so maybe the only thing to do is whitelist it.
> 
> Also, you now have a merge conflict in `src/control/server/ctl_harness.go`.

yes, "Cloneable" is Java keyword as a mark interface with which a object can be cloned easily without any user code.

**Comment by ashleypittman** [comment]
> yes, "Cloneable" is Java keyword as a mark interface with which a object can be cloned easily without any user code.

Thank you, I'll whitelist it and revert any changes to that file.

**Comment by ashleypittman** [comment]
> Still not passing checkpatch stage.

Interesting.  It seems that for this build there's a install/ and build/ dir in the tree which it's checking and failing on.  What's more there are fuse headers in the install tree, I thought we'd removed them.

I'll blacklist these directories and investigate.

File: Jenkinsfile [hunk 9]
```diff
@@ -330,7 +330,7 @@ pipeline {
                                    password: GITHUB_USER_PSW,
                                    ignored_files: "src/control/vendor/*:src/include/daos/*.pb-c.h:src/common/*.pb-c.[ch]:src/mgmt/*.pb-c.[ch]:src/iosrv/*.pb-c.[ch]:src/security/*.pb-c.[ch]:*.crt:*.pem:*_test.go:src/cart/_structures_from_macros_.h"
                         sh label: "codespell",
-			   script: "codespell --ignore-words-list nd,uint,ths,ba,creat,te,cas,mapp,pres,crashers,dout,tre,reord,mimick,cloneable --skip .git,*.png,*.PNG,*.pyc,./src/rdb/raft/*,./src/control/vendor/*,./.sconf_temp/*,.sconsign.dblite,RSA.golden,./code_review/*"
+			   script: "codespell --ignore-words-list nd,uint,ths,ba,creat,te,cas,mapp,pres,crashers,dout,tre,reord,mimick,cloneable --skip *.png,*.PNG,*.pyc,src/rdb/raft/*,src/control/vendor/*,RSA.golden `git ls-tree --full-tree --name-only HEAD`"
```
Comment (brianjmurrell) [inline Jenkinsfile:333]: I'm confused.  Doesn't daos-stack/code_review#21 do all of this right in check_patch?

Comment (ashleypittman) [reply Jenkinsfile:333]: Yes, that will replace this for new code.  This line will run the check for the current contents of the tree, so can be removed right before this PR lands.

File: src/tests/ftest/util/command_utils.py [hunk 10]
```diff
@@ -192,9 +192,16 @@ def stop(self):
...
+            # Use a list to send signals to the process with one second delays:
+            #   Interrupt + wait 3 seconds
+            #   Terminate + wait 2 seconds
+            #   Quit + wait 1 second
+            #   Kill
+            signal_list = [
+                signal.SIGINT, None, None, None,
+                signal.SIGTERM, None, None,
+                signal.SIGQUIT, None,
+                signal.SIGKILL]
```
Comment (phender) [inline src/tests/ftest/util/command_utils.py:204]: This is reverting a code change - something we don't want to do.

File: src/tests/ftest/util/daos_perf_utils.py [hunk 11]
```diff
@@ -36,7 +36,13 @@ def __init__(self, path):
...
 
         Args:
-            path (str): path to the daos_perf command
+            basepath (str): DAOS base path
+            processes (int): number of host processes
+            hostfile (str): file defining host names and slots
+            display (bool, optional): print daos_perf output to the console.
+                Defaults to True.
+        Raises:
+            DaosPerfFailed: if an error occurred running the daos_perf command
```
Comment (phender) [inline src/tests/ftest/util/daos_perf_utils.py:45]: This change does not match the method arguments.

File: Jenkinsfile [hunk 12]
```diff
@@ -329,6 +329,8 @@ pipeline {
                         checkPatch user: GITHUB_USER_USR,
                                    password: GITHUB_USER_PSW,
                                    ignored_files: "src/control/vendor/*:src/include/daos/*.pb-c.h:src/common/*.pb-c.[ch]:src/mgmt/*.pb-c.[ch]:src/iosrv/*.pb-c.[ch]:src/security/*.pb-c.[ch]:*.crt:*.pem:*_test.go:src/cart/_structures_from_macros_.h"
+                        sh label: "codespell",
+			   script: "codespell --ignore-words-list nd,uint,ths,ba,creat,te,cas,mapp,pres,crashers,dout,tre,reord,mimick,cloneable,keypair,bject,tread --builtin clear --skip *.png,*.PNG,*.pyc,src/rdb/raft/*,src/control/vendor/*,RSA.golden `git ls-tree --full-tree --name-only HEAD`"
```
[the remark this answers is not in the document]
Comment (brianjmurrell) [reply Jenkinsfile:350]: How do we get codespell not to check the spelling on the exception list?  It seems like every patch that updates the whitelist is itself going to fail the codespell.

Can the whitelist be put into a file, perhaps `cat`ted into the codespell commandline option and the whiltelist file excepted from codespell checking?

```suggestion
			   script: "codespell --ignore-words-list $(cat codespell-whitelist) --builtin clear --skip codespell-whitelist,*.png,*.PNG,*.pyc,src/rdb/raft/*,src/control/vendor/*,RSA.golden `git ls-tree --full-tree --name-only HEAD`"
```

Ultimately we **really** need to reduce the number of exceptions we are needing to allow through CI.  As you have seen, we are becoming complacent with allowing things to fail and land anyway and because of it we are introducing new failures that shouldn't have been allowed.

File: src/dtx/dtx_common.c [hunk 13]
```diff
@@ -303,19 +303,6 @@ dtx_leader_begin(struct dtx_id *dti, daos_unit_oid_t *oid, daos_handle_t coh,
...
-		 * background dedicated DTXs batched commit ULT has
-		 * not committed them in time. Let's retry later.
-		 */
-		D_DEBUG(DB_TRACE, "Too many potential conflict DTXs"
-			" for the given "DF_DTI", let's retry later.\n",
-			DP_DTI(dti));
-		D_FREE(dlh->dlh_subs);
-		return -DER_INPROGRESS;
-	}
-
```
Comment (brianjmurrell) [inline src/dtx/dtx_common.c]: Is this code removal intentional?  Seems like it's not related to adding codespell and fixing spelling errors.

File: src/tests/ftest/util/command_utils.py [hunk 14]
```diff
@@ -192,16 +192,9 @@ def stop(self):
...
-            #   Quit + wait 1 second
-            #   Kill
-            signal_list = [
-                signal.SIGINT, None, None, None,
-                signal.SIGTERM, None, None,
-                signal.SIGQUIT, None,
-                signal.SIGKILL]
+            # Send a SIGTERM to the stop the subprocess and if it is still
+            # running after 5 seconds send a SIGKILL and report an error
+            signal_list = [signal.SIGTERM, signal.SIGKILL]
```
Comment (brianjmurrell) [inline src/tests/ftest/util/command_utils.py]: This also seems unrelated, but more importantly, why the reduced list of signals.  `orterun` for example, AFAIU wants to be `SIGINT`ed to gracefully shut down.

File: Jenkinsfile (hunk 12 above)
[the remark this answers is not in the document]
Comment (ashleypittman) [reply Jenkinsfile:350]: Good questions.  Having the check here will check the entire tree but has really bad reporting, so what I think we should do is get this PR to a point where we're happy with the corrections, remove this change and then land it.  I was hopeful that once we had checking of PRs (which we've had for some weeks now) that we would no longer be introducing regressions, and that would be the end of it.

File: src/tests/ftest/util/command_utils.py (hunk 14 above)
Comment (ashleypittman) [reply src/tests/ftest/util/command_utils.py]: Unintentional, will fix.

File: src/dtx/dtx_common.c (hunk 13 above)
Comment (ashleypittman) [reply src/dtx/dtx_common.c]: Unintentional, will fix.

File: Jenkinsfile (hunk 12 above)
[the remark this answers is not in the document]
Comment (brianjmurrell) [reply Jenkinsfile:350]: I'm not sure I'm quite grasping

> remove this change 

means.  What would we be removing?

File: src/README.md [hunk 15]
```diff
@@ -30,11 +30,11 @@ As illustrated in the diagram below, a DAOS installation involves several compon
...
+A DAOS server is a multi-tenant daemon running on a Linux instance (i.e. physical node, VM or container) and managing the locally-attached NVM storage allocated to DAOS. It listens to a management port, addressed by an IP address and a TCP port number, plus one or more fabric endpoints, addressed by network URIs. The DAOS server is configured through a YAML file (i.e. /etc/daos_server.yml or different path provided on the command line) and can be integrated with different daemon management or orchestration frameworks (e.g. a systemd script, a Kunernetes service or even via a parallel launcher like pdsh or srun).
```
Comment (kccain) [inline src/README.md:33]: Kunernetes -> Kubernetes

File: src/cart/src/gurt/fault_inject.c [hunk 16]
```diff
@@ -631,15 +631,15 @@ d_fault_inject_is_enabled(void)
...
+ *                           support injecting X faults in Y occurrences
  */
 bool
 d_should_fail(struct d_fault_attr_t *fault_attr)
 {
 	bool			 rc = true;
 
 	if (!d_fi_initialized()) {
-		D_ERROR("fault injectiont not initalized.\n");
+		D_ERROR("fault injectiont not initialized.\n");
```
Comment (kccain) [inline src/cart/src/gurt/fault_inject.c:642]: injectiont -> injection

File: src/object/README.md [hunk 17]
```diff
@@ -108,7 +108,7 @@ forward shards embedded in the RPC request, when the leader shard gets that IO
 request it handles it as below steps:
 -   firstly forwards the IO request to others shards
     For the request forwarding, it is offload to the vos target's offload
-    extream to release the main IO service extream from IO request sending and
+    extreme to release the main IO service extreme from IO request sending and
```
Comment (kccain) [inline src/object/README.md:111]: xstream

File: Jenkinsfile [hunk 18]
```diff
@@ -351,8 +351,6 @@ pipeline {
                         checkPatch user: GITHUB_USER_USR,
                                    password: GITHUB_USER_PSW,
                                    ignored_files: "src/control/vendor/*:src/include/daos/*.pb-c.h:src/common/*.pb-c.[ch]:src/mgmt/*.pb-c.[ch]:src/iosrv/*.pb-c.[ch]:src/security/*.pb-c.[ch]:*.crt:*.pem:*_test.go:src/cart/_structures_from_macros_.h"
-                        sh label: "codespell",
-			   script: "codespell --ignore-words-list nd,uint,ths,ba,creat,te,cas,mapp,pres,crashers,dout,tre,reord,mimick,cloneable,keypair,bject,tread --builtin clear --skip *.png,*.PNG,*.pyc,src/rdb/raft/*,src/control/vendor/*,RSA.golden `git ls-tree --full-tree --name-only HEAD`"
```
Comment (brianjmurrell) [inline Jenkinsfile]: You don't want to have codespell always checking?

Comment (ashleypittman) [reply Jenkinsfile]: I do, but not using this technique.  In fact we already have codespell working on PRs, and reporting into tickets.

This line in the Jenkinsfile would check the entire tree which found a number of issues, however when doing so it would stop the entire build at the checkpatch stage, and would be really difficult to find the cause so the Jenkinsfile part of this PR was useful to find the issues but not good enough to be a part or our ongoing CI process.

Comment (brianjmurrell) [reply Jenkinsfile]: Ahh.  I see.
